commandapi-platforms/commandapi-paper/commandapi-paper-core/src/main/java/dev/jorel/commandapi/CommandAPIPaper.java

```diff
@@ -0,0 +1,244 @@
+		} else {
+			CommandAPI.logNormal("Did not hook into Paper ServerResourcesReloadedEvent while using commandapi-paper. Are you actually using Paper?");
+		}
+	}
+
+	private void checkPaperDependencies() {
+		try {
+			Class.forName("net.kyori.adventure.text.Component");
+			CommandAPI.logNormal("Hooked into Adventure for AdventureChat/AdventureChatComponents");
+		} catch (ClassNotFoundException e) {
+			if (CommandAPI.getConfiguration().hasVerboseOutput()) {
+				CommandAPI.logWarning("Could not hook into Adventure for AdventureChat/AdventureChatComponents");
+			}
+		}
+
+		isPaperPresent = false;
+
+		try {
+			Class.forName("io.papermc.paper.event.server.ServerResourcesReloadedEvent");
+			isPaperPresent = true;
+			CommandAPI.logNormal("Hooked into Paper for paper-specific API implementations");
+		} catch (ClassNotFoundException e) {
+			isPaperPresent = false;
+			if (CommandAPI.getConfiguration().hasVerboseOutput()) {
+				CommandAPI.logWarning("Could not hook into Paper for /minecraft:reload. Consider upgrading to Paper: https://papermc.io/");
+			}
+		}
+
+		isFoliaPresent = false;
+
+		try {
+			Class.forName("io.papermc.paper.threadedregions.RegionizedServerInitEvent");
+			isFoliaPresent = true;
+			CommandAPI.logNormal("Hooked into Folia for folia-specific API implementations");
+			CommandAPI.logNormal("Folia support is still in development. Please report any issues to the CommandAPI developers!");
+		} catch (ClassNotFoundException e) {
+			isFoliaPresent = false;
+		}
+	}
+
+	public CommandMap getCommandMap() {
+		return Bukkit.getCommandMap();
+	}
+
+	@Override
+	public Platform activePlatform() {
+		return Platform.PAPER;
+	}
+
+	@Override
+	public BukkitCommandSender<? extends CommandSender> wrapCommandSender(CommandSender sender) {
+		if (sender instanceof BlockCommandSender block) {
+			return new BukkitBlockCommandSender(block);
+		}
+		if (sender instanceof ConsoleCommandSender console) {
+			return new BukkitConsoleCommandSender(console);
+		}
+		if (sender instanceof Player player) {
+			return new BukkitPlayer(player);
+		}
+		if (sender instanceof org.bukkit.entity.Entity entity) {
+			return new BukkitEntity(entity);
+		}
+		if (sender instanceof NativeProxyCommandSender nativeProxy) {
+			return new BukkitNativeProxyCommandSender(nativeProxy);
+		}
+		if (sender instanceof ProxiedCommandSender proxy) {
+			return new BukkitProxiedCommandSender(proxy);
+		}
+		if (sender instanceof RemoteConsoleCommandSender remote) {
+			return new BukkitRemoteConsoleCommandSender(remote);
+		}
+		final Class<? extends CommandSender> FeedbackForwardingSender = feedbackForwardingCommandSender;
+		if (FeedbackForwardingSender.isInstance(sender)) {
+			// We literally cannot type this at compile-time, so let's use a placeholder CommandSender instance
+			return new BukkitFeedbackForwardingCommandSender<CommandSender>(FeedbackForwardingSender.cast(sender));
+		}
```

Comment:
Storing the class in a local variable (same for `NullCommandSender`) probably isn't necessary when the field is in the same class it is being used (previously the field came from `PaperImplementations` but was used in `CommandAPIBukkit`).

```suggestion
		if (this.feedbackForwardingCommandSender.isInstance(sender)) {
			// We literally cannot type this at compile-time, so let's use a placeholder CommandSender instance
			return new BukkitFeedbackForwardingCommandSender<CommandSender>(this.feedbackForwardingCommandSender.cast(sender));
		}
```